qemu/tests/migration_multi_host_with_stress.py

```diff
@@ -0,0 +1,62 @@
+import logging
+import time
+from autotest.client.shared import utils, error
+from virttest import utils_test
+from virttest import utils_misc
+
+
+@error.context_aware
+def run(test, params, env):
+    """
+    KVM multi-host migration test:
+
+    Migration execution progress is described in documentation
+    for migrate method in class MultihostMigration.
+    steps:
+        1) boot vm and login, load stress in guest
+        2) do migration
+        3) wait for a while and stop stress 
+        4) login guest when migrate done
+
+    :param test: kvm test object.
+    :param params: Dictionary with test parameters.
+    :param env: Dictionary with the test environment.
+    """
+    preprocess_env = params.get("preprocess_env", "yes") == "yes"
+    mig_protocol = params.get("mig_protocol", "tcp")
+    mig_type = utils_test.qemu.MultihostMigration
+    if mig_protocol == "fd":
+        mig_type = utils_test.qemu.MultihostMigrationFd
+    if mig_protocol == "exec":
+        mig_type = utils_test.qemu.MultihostMigrationExec
+    if "rdma" in mig_protocol:
+        mig_type = utils_test.qemu.MultihostMigrationRdma
+
+    class TestMultihostMigration(mig_type):
+
+        def __init__(self, test, params, env):
+            super(TestMultihostMigration, self).__init__(test, params, env,
+                                                         preprocess_env)
+            self.srchost = self.params.get("hosts")[0]
+            self.dsthost = self.params.get("hosts")[1]
+            self.vm = params["vms"].split()[0]
+            self.login_timeout = int(params.get("login_timeout", 360))
+
+        def migration_scenario(self, worker=None):
+            def worker(mig_data):
+                logging.info("Try to login guest before migration test.")
+                vm = env.get_vm(params["main_vm"])
+
+                session = vm.wait_for_login(timeout=self.login_timeout)
+
+                logging.info("Do stress test before migration.")
+                stress_cmd = params.get("stress_cmd")
+                session.sendline(stress_cmd)
+
+            if params.get("check_vm_before_migration", "yes") == "no":
+                params["check_vm_needs_restart"] = "no"
+
+            self.migrate_wait([self.vm], self.srchost, self.dsthost, worker)
+
+    mig = TestMultihostMigration(test, params, env)
+    mig.run()
```

Comment:
Hi @spcui 

For this case, how about the status checking after migration?

Thanks.